### PR TITLE
[INT-14314]Conditionally render textInput in and update border radius in Filter Button in Filter Controller

### PR DESCRIPTION
**Pull request by briancjkim**
intellihr/engineering/team-kiki/team-kiki/-/issues/14314
hiddensearchbar
**The following MUST be checked prior to approval. If not relevant to your PR, remove the line from your description.**
- [ ]  The `styleguide.config.js` has been updated to show examples/new functionality for the component
- [ ]  Test Coverage for any new or updated functionality
- [ ]  New and updated components have been tested locally in lapis and SPA and work as expected
- [ ]  This PR has been tagged with PATCH, MINOR, or MAJOR.
- [ ]  The component has data-component-type and data-component-context attributes following the standard
- [ ]  You have requested a cross-team review on this component so everyone knows it exists

### CODE REVIEW

File: src/domain/Filters/FilterController/FilterController.tsx [hunk 1]
```diff
@@ -120,4 +134,17 @@ export class FilterController extends React.PureComponent<IFilterControllerProps
       </InputGroup.Button>
     )
   }
+
+  private filterButtonWithRoundBorderAllSide = ({ toggleMenu, toggleComponentRef, ariaProps }: any) => {
+    return (
+      <InputGroup.Button
```
Comment (wyvl) [inline src/domain/Filters/FilterController/FilterController.tsx:140]: Should be just Button?

File: src/domain/Inputs/InputGroup/subcomponents/style.ts [hunk 2]
```diff
@@ -36,6 +36,8 @@ const InputGroupButton = styled.button<IInputGroupButtonProps>`
         return css`border-radius: 4px 0 0 4px !important;`
       case 'right':
         return css`border-radius: 0 4px 4px 0 !important;`
+      case 'middle':
+        return css`border-radius: 4px 4px 4px 4px !important;`
```
Comment (andrewhillcode) [inline src/domain/Inputs/InputGroup/subcomponents/style.ts:40]: A 'middle' style in an input group would actually have no border radius 😅 

Also we really shouldn't be using the InputGroup.Button component at all outside of an InputGroup

File: src/domain/Filters/FilterController/FilterController.tsx [hunk 3]
```diff
@@ -120,4 +134,17 @@ export class FilterController extends React.PureComponent<IFilterControllerProps
...
+        onClick={toggleMenu}
+        innerRef={toggleComponentRef}
+        groupPosition='middle'
```
Comment (wyvl) [inline src/domain/Filters/FilterController/FilterController.tsx:143]: then you don't need this if button

File: src/domain/Inputs/InputGroup/subcomponents/style.ts [hunk 4]
```diff
@@ -36,6 +36,8 @@ const InputGroupButton = styled.button<IInputGroupButtonProps>`
         return css`border-radius: 4px 0 0 4px !important;`
       case 'right':
         return css`border-radius: 0 4px 4px 0 !important;`
+      case 'middle':
```
Comment (wyvl) [inline src/domain/Inputs/InputGroup/subcomponents/style.ts:39]: don't need to change input group

File: src/domain/Inputs/InputGroup/__snapshots__/InputGroup.test.tsx.snap [hunk 5]
```diff
@@ -6,6 +6,10 @@ exports[`<InputGroup /> should render a simple input group 1`] = `
     groupPosition="left"
     name="test-input"
   />
+  <TextInput
```
Comment (wyvl) [inline src/domain/Inputs/InputGroup/__snapshots__/InputGroup.test.tsx.snap:9]: don't need to change input group

File: src/domain/Inputs/InputGroup/InputGroup.test.tsx [hunk 6]
```diff
@@ -9,6 +9,7 @@ describe('<InputGroup />', () => {
     const wrapper = shallow(
       <InputGroup>
         <TextInput name='test-input' groupPosition='left' />
+        <TextInput name='test-input' groupPosition='middle' />
```
Comment (wyvl) [inline src/domain/Inputs/InputGroup/InputGroup.test.tsx:12]: don't need to change input group

File: src/domain/Filters/FilterController/FilterController.tsx (hunk 1 above)
Comment (andrewhillcode) [inline src/domain/Filters/FilterController/FilterController.tsx:140]: I think you could just use a regular `<Button>` component here

File: src/domain/Filters/FilterController/FilterController.tsx [hunk 7]
```diff
@@ -120,4 +134,17 @@ export class FilterController extends React.PureComponent<IFilterControllerProps
       </InputGroup.Button>
     )
   }
+
+  private filterButtonWithRoundBorderAllSide = ({ toggleMenu, toggleComponentRef, ariaProps }: any) => {
```
Comment (andrewhillcode) [inline src/domain/Filters/FilterController/FilterController.tsx:138]: can probably rename this function to `filterButton` and the other one to `inputGroupFilterButton`

**Review by andrewhillcode — APPROVED** [review]

**Pull request merged**